### PR TITLE
[IMP] account: enable account type domain filter using the context

### PR DESCRIPTION
**Pull request by h4818**
Users are able to create accounts without restriction on the type of account to be created. With this commit we provide a context key to limit available account types and thus minimise errors.

part of task-2684816

### CODE REVIEW

File: addons/account/static/src/js/account_selection.js [hunk 1]
```diff
@@ -58,7 +67,7 @@ odoo.define('account.hierarchy.selection', function (require) {
         },
         _addHierarchy: function(el, group, level) {
             var self = this;
-            _.each(group, function(item) {
+            _.each(_.compact(group), function(item) {
```
Comment (william-andre) [inline addons/account/static/src/js/account_selection.js:70]: Why is this needed? When are you adding falsy elements in `self.hierarchy_groups`?

File: addons/account/static/src/js/account_selection.js [hunk 2]
```diff
@@ -14,39 +14,48 @@ odoo.define('account.hierarchy.selection', function (require) {
...
+                            'expense': {'parent': _t('Profit & Loss'), 'name': _t('Expense'), 'index': 1},
+                        }
+                        _.each(unique_groups, g => {
+                            if (g in lookup){
+                                var rec = lookup[g]
+                                if (self.hierarchy_groups[rec['index']]) {
+                                    self.hierarchy_groups[rec['index']].children.push({'name': rec['name'], 'ids': _.map(_.filter(arg, v => v['internal_group'] == g), v => v['id'])});
+                                } else {
+                                    self.hierarchy_groups[rec['index']] = {'name': rec['parent'], 'children': [{'name': rec['name'], 'ids': _.map(_.filter(arg, v => v['internal_group'] == g), v => v['id'])}]};
+                                }
```
[the remark this answers is not in the document]
Comment (h4818) [reply addons/account/static/src/js/account_selection.js]: ok

File: addons/account/static/src/js/account_selection.js [hunk 3]
```diff
@@ -14,39 +14,48 @@ odoo.define('account.hierarchy.selection', function (require) {
...
+                            } else {
+                                self.hierarchy_groups[2] = {'name': _t('Other'), 'ids': _.map(_.filter(arg, v => !['asset', 'liability', 'equity', 'income', 'expense'].includes(v['internal_group'])), v => v['id'])};
```
[the remark this answers is not in the document]
Comment (h4818) [reply addons/account/static/src/js/account_selection.js]: ok

File: addons/account/static/src/js/account_selection.js [hunk 4]
```diff
@@ -14,39 +14,48 @@ odoo.define('account.hierarchy.selection', function (require) {
...
+                    self.hierarchy_groups = [];
+                    var unique_groups = _.uniq(_.map(arg, v => v['internal_group']));
+                    if (unique_groups.length > 1) {
+                        var lookup = {
+                            'asset': {'parent': _t('Balance Sheet'), 'name': _t('Assets'), 'index': 0},
+                            'liability': {'parent': _t('Balance Sheet'), 'name': _t('Liabilities'), 'index': 0},
+                            'equity': {'parent': _t('Balance Sheet'), 'name': _t('Equity'), 'index': 0},
+                            'income': {'parent': _t('Profit & Loss'), 'name': _t('Income'), 'index': 1},
+                            'expense': {'parent': _t('Profit & Loss'), 'name': _t('Expense'), 'index': 1},
+                        }
```
[the remark this answers is not in the document]
Comment (h4818) [reply addons/account/static/src/js/account_selection.js]: ok

File: addons/account/static/src/js/account_selection.js (hunk 1 above)
Comment (h4818) [reply addons/account/static/src/js/account_selection.js]: no falsy elements added now

**Comment by william-andre** [comment]
@robodoo r+
